### PR TITLE
Prevent applying boolean ops to `NotImplemented`

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Steps

- [ ] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description
Prevents this DeprecationWarning:

```DeprecationWarning: NotImplemented should not be used in a boolean context```

Reproducer:
``` python -Werror -m pylint ipaddress```
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

### CODE REVIEW

File: tests/unittest_inference.py [hunk 1]
```diff
@@ -1117,6 +1117,12 @@ def __radd__(self, other): return NotImplemented
         first = next(ast_node.infer())
         self.assertEqual(first, util.Uninferable)
 
+    @pytest.mark.filterwarnings("error::DeprecationWarning")
```
Comment (Pierre-Sassoulas) [inline tests/unittest_inference.py:1120]: Do we need to filter that ? I thought the fix was precisely to remove the deprecation warning ?

**Review by Pierre-Sassoulas — APPROVED** [review]

File: tests/unittest_inference.py (hunk 1 above)
Comment (jacobtylerwalls) [reply tests/unittest_inference.py:1120]: Good Q, "filter" is a little misleading here. It's taking DeprecationWarning and raising it to `error`. Revert the fix and this test will fail.

Comment (jacobtylerwalls) [reply tests/unittest_inference.py:1120]: Otherwise the test would pass anyway, since `pytest` doesn't fail tests when there are deprecationwarnings.

Comment (Pierre-Sassoulas) [reply tests/unittest_inference.py:1120]: We might as well change the global configuration, we often have to so "fix deprecation warning" merge request both for astroid and pylint.

**Pull request merged**